File: solutions/euphoria/hamming/1/hamming.ex
public function distance(sequence left, sequence right)
    if length(left) != length(right) then
        return "left and right strands must be of equal length"
    end if

    integer hamming_distance = 0
    for i = 1 to length(left) do
        if left[i] != right[i] then
            hamming_distance += 1
        end if
    end for

    return hamming_distance
end function

File: solutions/delphi/hamming/1/hamming.pas
unit uHamming;

interface

type
  THamming = class
    class function Distance(const AChain1, AChain2:string):integer;
  end;

implementation

uses
  System.SysUtils;

{ THamming }

class function THamming.Distance(const AChain1, AChain2: string): integer;
var
  i: Integer;
begin
  if (AChain1=string.Empty) and (AChain2<>string.Empty) then
    raise EArgumentException.Create('error: left strand must not be empty');

  if (AChain2=string.Empty) and (AChain1<>string.Empty) then
    raise EArgumentException.Create('error: right strand must not be empty');

  if Length(AChain1) <> Length(AChain2) then
    raise EArgumentException.Create('error: left and right strands must be of equal length');

  Result := 0;
  for i := Low(AChain1) to High(AChain1) do
    if (AChain1[i] <> AChain2[i]) then
      Inc(Result);
end;

end.